City states button influence bar

r3versi opened the pull request:
![influence_bar](https://user-images.githubusercontent.com/17114100/95583479-79a36900-0a3c-11eb-8c94-7a9e5f4d17f7.png)

I also changed the thresholds of influence for city states, because the state `Unforgivable` wasn't "reachable" in practice:
* `< 30 || atWar` -> `Unforgivable`
* `<0` -> `Enemy`
* `< 30` -> `Neutral` 
* `< 60` -> `Friend`
* `>60 && allied` -> `Ally`

yairm210 commented on core/src/com/unciv/ui/utils/ImageGetter.kt:
```diff
@@ -256,6 +258,63 @@ object ImageGetter {
         return advancementGroup
     }
 
+    fun getInfluenceBar(influence: Float, relationshipLevel: RelationshipLevel, influenceBarSize: Float = 100f): Table {
```
This is *only* used by the city button, so it should be a function in the city button

r3versi commented:
I was thinking of adding it to the diplomacy screen (with slightly
different width and height). What do you think?

On Fri, Oct 9, 2020, 15:34 Yair Morgenstern <notifications@github.com>
wrote:

> *@yairm210* commented on this pull request.
> ------------------------------
>
> In core/src/com/unciv/ui/utils/ImageGetter.kt
> <https://github.com/yairm210/Unciv/pull/3248#discussion_r502430719>:
>
> > @@ -256,6 +258,63 @@ object ImageGetter {
>          return advancementGroup
>      }
>
> +    fun getInfluenceBar(influence: Float, relationshipLevel: RelationshipLevel, influenceBarSize: Float = 100f): Table {
>
> This is *only* used by the city button, so it should be a function in the
> city button
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/yairm210/Unciv/pull/3248#pullrequestreview-505677385>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AECSH5DGZKR7K25MA5TKLQLSJ4GPNANCNFSM4SKAZ2PA>
> .
>

yairm210 commented:
I still feel that the ImageGetter is getting very clogged.
If that's the case, then I would still advise that the function goes in the CityButton, but in a companion object

r3versi commented:
I see, but consider that ImageGetter is slim wrt to other classes. Anyway
I'm moving it to the CityButton although that's not the "semantically"
right place imho.

r3versi commented:
OK, moved the method to CityButton and added it to DiplomacyScreen. Here is a preview
![Screenshot from 2020-10-09 15-50-54](https://user-images.githubusercontent.com/17114100/95591245-56ca8200-0a47-11eb-9ad4-3323d9fb2112.png)
